### PR TITLE
Add FXIOS-7944 [v122] Add Webview telemetry for failures

### PR DESCRIPTION
**Pull request by lmarceau**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7944)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17716)

## :bulb: Description
Add Web View telemetry for failures + some logs related to JavaScript and failure points that can help us debug some crashes in Sentry.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by lmarceau** [comment]
@dnarcese 

# Request for data collection review form

1) What questions will you answer with this data?
These will be used as part of a series of metrics to monitor the overall health of the app. We want to get a baseline on webview errors, so when we release the webview refactor we can determine if there's any spikes in those errors.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
If we see a spike in the number of webview errors it's a strong signal that something has broken.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
There are no other methods

4) Can current instrumentation answer these questions?
No

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Recorded when an error occurs while starting to load data for the main frame, so early navigation.</td>
    <td>Category 1</td>
    <td>https://mozilla-hub.atlassian.net/browse/FXIOS-7944</td>
  </tr>
  <tr>
    <td>Recorded when an error occurred during navigation.</td>
    <td>Category 1</td>
    <td>https://mozilla-hub.atlassian.net/browse/FXIOS-7944</td>
  </tr>
  <tr>
    <td>Recorded when an error page is shown on the webview.</td>
    <td>Category 1</td>
    <td>https://mozilla-hub.atlassian.net/browse/FXIOS-7944</td>
  </tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
https://dictionary.telemetry.mozilla.org/apps/firefox_ios

7) How long will this data be collected?  Choose one of the following:
Forever, it will be used as ongoing health metrics

8) What populations will you measure?

* Which release channels?
All

* Which countries?
All

* Which locales?
All

9) If this data collection is default on, what is the opt-out mechanism for users?
The user can opt out

10) Please provide a general description of how you will analyze this data.
Glean

11) Where do you intend to share the results of your analysis?
The iOS team

12) Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection? If so:
No third party tools

File: firefox-ios/Client/metrics.yaml [hunk 1]
```diff
@@ -4914,6 +4914,46 @@ app_errors:
       - fx-ios-data-stewards@mozilla.com
     expires: never
 
+webview:
+  did_fail_provisional:
+    type: event
+    description: |
+      Recorded when an error occurs while starting to load data for the main frame, so early navigation.
```
Comment (dnarcese) [inline firefox-ios/Client/metrics.yaml:4921]: I don't really understand this description. Can you reword it

**Comment by dnarcese** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This will be reviewed by me every 6 months

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Type 2

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

No

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

**Comment by lmarceau** [comment]
@Mergifyio backport release/v122

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**